### PR TITLE
docs: update the definition of higher order function

### PR DESCRIPTION
**Pull request by Daggron**
- Update the note for higher order function to represent a complete definition of higher order functions

### Description

- Updated the side note for Higher order functions which was representing incomplete definition.

### Motivation

- Devs were confusing here that a function which returns another function can only be called a higher order function, which is incomplete, The complete definition of HOF is `a function which either takes other functions as argument or return a function is called an HOF`

Same Definition here -> https://eloquentjavascript.net/05_higher_order.html#h_xxCc98lOBK

### CODE REVIEW

File: files/en-us/glossary/first-class_function/index.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ function sayHello() {
 
 In this example, we are returning a function from another function - _We can return a function because functions in JavaScript are treated as values._
 
-> **Note:** A function that returns a function is called a _higher-order function_.
+> **Note:** A function that returns a function or take other function as arguments and operates over them is called a _higher-order function_.
```
Comment (Josh-Cena) [inline files/en-us/glossary/first-class_function/index.md:57]:
```suggestion
> **Note:** A function that returns a function or takes other function as arguments is called a _higher-order function_.
```

Comment (Daggron) [reply files/en-us/glossary/first-class_function/index.md]: updated

File: files/en-us/glossary/first-class_function/index.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ function sayHello() {
 
 In this example, we are returning a function from another function - _We can return a function because functions in JavaScript are treated as values._
 
-> **Note:** A function that returns a function is called a _higher-order function_.
+> **Note:** A function that returns a function or takes other function as arguments is called a _higher-order function_.
```
Comment (Josh-Cena) [inline files/en-us/glossary/first-class_function/index.md:57]:
```suggestion
> **Note:** A function that returns a function or takes other functions as arguments is called a _higher-order function_.
```

**Review by Josh-Cena — APPROVED** [review]
Thanks, and congratulations on landing your first PR here—welcome aboard!

**Comment by Daggron** [comment]
> Thanks, and congratulations on landing your first PR here—welcome aboard!

Thankyou 🎉

**Pull request merged**